README.md

```diff
@@ -439,3 +451,32 @@ triggers the next parsing element through the `parserMainLoop` function. This
 approach not only adds an extra layer of control to the parser along execution steps, but
 also enables parsing of customized RDB files or even specific parts of the file. This
 functionality can be further enhanced as needed.
+
+## Troubleshooting
+This section provides assistance for diagnosing and debugging Redis server failures that
+may occur when attempting to stream RESP commands using the parser. Analyzing these failures
+can be particularly challenging in various scenarios:
+
+- **Pipeline Mode**: When using pipeline mode, which involves multiple concurrent pending 
+commands at any given moment.
+
+- **Empty Redis Server**: When the `delKeyBeforeWrite` flag is not used, and the Redis 
+server is not empty.
+
+- **Production Env**: In a production environment where reliability and efficiency 
+are critical.
+
+
+One approach can be to have a dry-run by assisting `RDBX_createRespToFileWriter` and dump
+RESP commands into file for investigation. The following debug functions are designed to 
+further help with the investigation:
+
+### Function RDBX_enumerateCmds
+Enumerate commands by preceding any RESP command with an additional trivial RESP command 
+of the type `echo <cmd-number>`. This can be especially useful since the RESP-to-Redis 
+instance prints the problematic command ID (and command prefix) in case of a failure.
```

Comment:
I was thinking about adding `SET __rdb_cmd_number <num>` but then I thought maybe i shouldn't modify the db with additional data. But now that you suggest it as well, maybe it is actually a good idea :)

Reply:
if it's just for debugging, it may be a good idea.. if it's also for production troubleshooting it may be an issue, but since it's optional, i suppose that's fine.
either way, i still don't understand how the ECHO thing used to work for you.

Reply:
I could have dumped enumerated RESP protocol commands. I could have played it and capture it in network analyzer or recorded in redis without any side effect. Anyway, I think `SET __rdb_cmd_number <num>` is more user friendly.